llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -54,9 +73,15 @@ TGLexer::TGLexer(SourceMgr &SM, ArrayRef<std::string> Macros) : SrcMgr(SM) {
   PrepIncludeStack.push_back(
       std::make_unique<std::vector<PreprocessorControlDesc>>());
 
-  // Put all macros defined in the command line into the DefinedMacros set.
-  for (const std::string &MacroName : Macros)
+  // Add all macros defined on the command line to the DefinedMacros set.
+  // Check invalid macro names and print fatal error if we find one.
+  for (const std::string &MacroName : Macros) {
+    if (!IsValidMacroName(MacroName))
+      PrintFatalError(Twine("Invalid macro name `") + Twine(MacroName) +
+                      Twine("` specified on command line."));
```

Comment:
With the above suggestion applied you don't need Twines here.

Reply:
N/A anymore.

Reply:
Done now.

Reply:
I mean you don't need Twines at all. This is just a nitpick though, so feel free to ignore.

Reply:
I see what you mean. I'll incorporate it, no worries.

Reply:
Done as well.